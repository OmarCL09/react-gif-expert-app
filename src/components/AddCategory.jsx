import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onAddCategory }) => {

    const [inputValue, setInputValue] = useState('One Punch Man');

    const onInputChange = ( e ) => {
        e.preventDefault();
        setInputValue( e.target.value );
    }

    const onSubmit = ( e ) => {
        e.preventDefault();
        if(inputValue.trim().length <= 1) return;
        setInputValue('');
        onAddCategory( inputValue.trim());
    }

  return (
    <form onSubmit={ onSubmit } aria-label="form">
        <input 
            type="text" 
            placeholder="Search some gifs"
            value={ inputValue }
            onChange={ onInputChange }
        />
    </form>
  )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
}
