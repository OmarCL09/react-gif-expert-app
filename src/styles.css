@import url('https://fonts.googleapis.com/css2?family=Katibeh&family=Poppins:wght@300&display=swap');

* {
  font-family: 'Poppins', sans-serif;
  background-color: rgb(234, 234, 234);
}
  
body {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  text-align: center;
}
body::-webkit-scrollbar{
  display: none;
}
  
input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 70%;
}
  
h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin: 0;
  margin-right: 1%;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;  
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;
}

.card p {
  background-color: white;
  flex: 1;
  font-size: 1.5rem;
  margin-top: 5px;
  padding: 5px 20px 0px 20px;
  text-align: center;
}

.card img {
  width: 100%;
}
.button-test{
  background: none;
  font-size: 1.3em;
  margin-top: 1%;
  padding: 1% 2%;
  border-radius: 15px;
  border: 1px solid black;
  cursor: pointer;
  transition: .2s;
}
.button-test:hover{
  transform: scale(0.98);
}
.title-container{
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: center;
  margin: 1% 0;
}
.buttonToDelete{
  width: 150px;
  padding: 0.5%;
  border: 1px solid black;
  border-radius: 20px;
  cursor: pointer;
  transition: .2s;
}
.buttonToDelete:hover{
  background-color: gray;
  transform: scale(0.98);
  color: white;
}
@media (max-width: 600px) {
  h3{
    font-size: 2em;
  }
  .title-container{
    flex-direction: column;
  }
}