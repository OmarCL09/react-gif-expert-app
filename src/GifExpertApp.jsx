import React, { useState } from 'react'
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch Man']);

    const onAddCategory = ( newCategory ) => {
        if( categories.includes(newCategory)) return;
        
        setCategories([ newCategory, ...categories ]);
    }

    const onDeleteCategory = (categorieToDelete) => {
        setCategories(categories.filter( category => category !== categorieToDelete));
    }

  return (
    <>

        <h1>GifExpertApp</h1>

        <AddCategory onAddCategory = { onAddCategory }/>

        {
            categories.map( ( category ) => (
                <GifGrid key={ category } category={ category } onDeleteCategory={ onDeleteCategory }/>
            ))
        }
    </>
  ) 
}
