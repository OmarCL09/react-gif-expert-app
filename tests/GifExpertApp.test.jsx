import { render, screen, fireEvent } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';
import { AddCategory } from "../src/components/AddCategory";

describe('Pruebas en GifExpertApp', () => {

    test('Debe estar solamente One Punch Man como default en categories', () => {
        
        const { container } = render( <GifExpertApp />);
        expect( container ).toMatchSnapshot();
        expect( screen.getAllByText('One Punch Man'));

    });

    test('Se debe agregar una nueva categoria al DOM', () => { 

        const { container } = render( <GifExpertApp /> );
        const input = screen.getByRole('textbox');
        const form  = screen.getByRole('form');
        
        fireEvent.input( input, { target: { value: 'Chainsaw man' } });
        fireEvent.submit( form );

        expect( container ).toMatchSnapshot();
        expect( screen.getAllByText('Chainsaw man'));

    })
});