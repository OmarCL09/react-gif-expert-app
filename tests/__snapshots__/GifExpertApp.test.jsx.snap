// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe estar solamente One Punch Man como default en categories 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search some gifs"
      type="text"
      value="One Punch Man"
    />
  </form>
  <div
    class="title-container"
  >
    <h3>
      One Punch Man
    </h3>
    <button
      class="buttonToDelete"
    >
      Delete
    </button>
  </div>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;

exports[`Pruebas en GifExpertApp Se debe agregar una nueva categoria al DOM 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search some gifs"
      type="text"
      value=""
    />
  </form>
  <div
    class="title-container"
  >
    <h3>
      Chainsaw man
    </h3>
    <button
      class="buttonToDelete"
    >
      Delete
    </button>
  </div>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
  <div
    class="title-container"
  >
    <h3>
      One Punch Man
    </h3>
    <button
      class="buttonToDelete"
    >
      Delete
    </button>
  </div>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
